src/lib/utils/size_estimation_utils.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <random>
+
+#include "types.hpp"
+
+namespace opossum {
+
+// In case the initially preallocated capacity is too small, the string is allocated on the heap.
+template <typename T>
+size_t memory_usage_of_string_objects(const T& string_vector) {
```

Comment:
If this always takes a `string_vector`, why is this templated?

Reply:
Depending on the caller, it might be a `pmr_vector` or `std::vector` with a `pmr_string` or `std::string`.